driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -139,7 +138,7 @@ public boolean hasTimeoutMS() {
      * @param onExpired the runnable to run
      */
     public void onExpired(final Runnable onExpired) {
-        Timeout.nullAsInfinite(timeout).onExpired(onExpired);
+        Timeout.nullAsInfinite(timeout).onExpired(onExpired, MILLISECONDS);
```

Comment:
The unit is changed to MILLISECONDS to align with other logic in the `TimeoutContext`, which also calculates in MILLISECONDS.